### PR TITLE
Add component log storage + an API to access logs 

### PR DESCRIPTION
**Pull request by Others**
This needs to be documented in the spec. Also needs supporting code in the deployment manager

### CODE REVIEW

File: src/component/logs.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                Ok(None)
+            } else {
+                self.policy_handle.get_contents().map(Some)
+            },
+        )
+    }
+}
+
+#[derive(Debug)]
+pub enum LogPolicy {
```
Comment (pcodes) [inline src/component/logs.rs]: Is my understanding correct that the LogPolicy is basically how you handle log output, so either sending it to a file, or having it be ignored?

File: src/component/logs.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    ToFile(NamedTempFile),
+    // Literally everywhere you might have a LogPolicy, having an Ignore policy is valid
+    // Thus we incorporate it into the struct itself, rather than everyone using `Option<LogPolicy>`
+    Ignore,
+}
+
+impl LogPolicy {
+    pub fn new_no_op_policy() -> Arc<Self> {
```
Comment (pcodes) [inline src/component/logs.rs]: no_op policy?

File: src/model.rs [hunk 3]
```diff
@@ -79,6 +79,21 @@ pub struct DeactivateResponse {
...
+    pub id: ComponentId,
+
+    pub dedup_number: u64,
+    pub log: Option<String>,
+
+    pub error: Option<String>,
+}
+
+#[derive(Clone, Deserialize, Debug, Eq, Hash, PartialEq, Serialize)]
+pub struct LogResponse {
```
Comment (pcodes) [inline src/model.rs]: What is a LogResponse used for?

File: src/component/logs.rs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+impl LogTracker {
+    pub fn new() -> Self {
+        Self {
+            dedup_number: DEDUP_COUNTER.fetch_add(1, Ordering::SeqCst),
+            policy_handle: LogPolicy::new_no_op_policy(),
+        }
+    }
+
+    pub fn create_associated_policy(&mut self) -> Result<Arc<LogPolicy>, WorkerError> {
```
Comment (pcodes) [inline src/component/logs.rs]: So does this just set the policy to be "write to a file"?

File: src/component/logs.rs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        match self {
+            Self::ToFile(f) => {
+                // We don't use the internal `File`, since that may have a cursor in any location
+                Ok(read_to_string(f.path())?)
+            }
+            Self::Ignore => Ok(String::new()),
+        }
+    }
+
+    pub fn get_popen_config(&self) -> Result<PopenConfig, WorkerError> {
```
Comment (pcodes) [inline src/component/logs.rs]: Why is this in log policy?

File: src/component/logs.rs (hunk 1 above)
Comment (Others) [reply src/component/logs.rs]: Yep!

File: src/component/logs.rs (hunk 2 above)
Comment (Others) [reply src/component/logs.rs]: Changed name to `ignored_policy`

File: src/model.rs (hunk 3 above)
Comment (Others) [reply src/model.rs]: It's a JSON object returned by the call to `/meta/logs`

File: src/component/logs.rs (hunk 4 above)
Comment (Others) [reply src/component/logs.rs]: The bit that's doing the logging just gets a log policy. An associated policy is a file writing policy that the `LogTracker` knows about, and can use to service requests to `logs`

File: src/component/logs.rs (hunk 5 above)
Comment (Others) [reply src/component/logs.rs]: For convenience. All the log policy is used for is creating `PopenConfig`s that write to the correct file

**Pull request merged**